src/providers/Wallet.js

```diff
@@ -74,43 +111,9 @@ function WalletAugmented({ children }) {
   }, [account, connector, wallet])
 
   const resetConnection = useCallback(async () => {
-    setChainId(-1)
     await wallet.reset()
   }, [wallet])
 
-  // Handle connect automatically if an account is available and we have some connected address on the wallet
-  useEffect(() => connect(), []) // eslint-disable-line
-
-  // This useEffect is needed because we don't have inmediatly available wallet.chainId  right after connecting in the previous hook
-  useEffect(() => {
-    if (wallet.account != null && chainId !== wallet.chainId) {
-      setChainId(wallet.chainId)
-      if (SUPPORTED_CHAINS.includes(wallet.chainId)) {
-        setPreferredChain(wallet.chainId)
-        setPreferredNetwork(wallet.chainId)
-      }
-    }
-  }, [wallet.account, wallet.chainId, chainId])
-
-  /* Note that with this hook and the previous one we are manually reseting the connection on chain changed detected.
-  Here i noticed that is better to always depend on the data that comes from the use-wallet library instead  of just subscribing to window.ethereum.on('chainChanged', handleChainChanged) 
-  because for some reason the reconnect action is way faster and the windows event subscription was triggering many re renders */
-  useEffect(() => {
```

Comment:
Note that this effect is not needed as the `connect` function from useWallet already [resets](https://github.com/aragon/use-wallet/blob/d431fdec8e8f7b47867f74469bef900ff19423f4/src/index.tsx#L199) the connection.